### DISCUSSION TITLE
Is there any example of loading a fbx model or any other 3d model?

### DISCUSSION THREAD

**Discussion by asheesh1996** [dotnet/Silk.NET#619] [Q&A]
I was able to setup a simple environment modifying cube to a plane and setup lights and camera with controls using the examples from https://github.com/dotnet/Silk.NET/tree/main/src/Lab/Experiments and https://github.com/dotnet/Silk.NET/tree/main/examples/CSharp.

**Comment by Perksey** [comment]
There are no examples of model loading yet. We provide Silk.NET.Assimp which has a one-to-one mapping of Assimp's C API, but I appreciate that this is relatively unused compared to Assimp's C++ API.

Here's an example of (a very old version) Silk.NET.Assimp being used "in the field": https://github.com/john-h-k/Voltium/blob/master/sources/Voltium.ModelLoader/ModelLoader.cs#L151, but beyond that there's not many resources I can point you to as we haven't had the time to create any yet.

A welcome community contribution would be putting together a demo in `examples/CSharp/OpenGL Demos/AssimpOpenGL`, will open an issue for this now.

**Comment by ponahoum** [comment]
@Perksey another good and simple example would be to convert a 3D model from a format to another format (+ applying useful processing import operations such as aiProcess_JoinIdenticalVertices etc.). Let's say from FBX or OBJ to GLTF. 
This is a very common operation that a lot of people would need in .Net, and if they do not know Marshaling/how native bindings work, a sample code would be useful.

I sometimes use Assimp.Net but it is unmaintained and buggy, I'm looking for a viable and maintained alternative and Silk seems a pretty good one so far, it just lacks comprehensive C# binding for Assimp.

**Reply by Perksey** [reply]
> I'm looking for a viable and maintained alternative and Silk seems a pretty good one so far, it just lacks comprehensive C# binding for Assimp.

There's an issue open for this and has been for a while - we're fine with the low-level bindings so higher-level ones will need community contribution: #249

**Comment by abenedik** [comment]
A working example of using Silk.Net.Assimp is also Ab3d.DXEngine.Assimp library that loads 3D objects for Ab3d.DXEngine (DirectX 11 rendering engine for .Net). You can see the full source for the Ab3d.DXEngine.Assimp and a usage sample project on GitHub: [https://github.com/ab4d/Ab3d.DXEngine.Assimp](https://github.com/ab4d/Ab3d.DXEngine.Assimp)
